### DISCUSSION TITLE
properly override toJSON for mathjs bignumber

### DISCUSSION THREAD

**Discussion by ORESoftware** [josdejong/mathjs#2849] [Q&A]
I am trying to override mathjs Bignumber using:

```js
export const bgn = (v: number | math.BigNumber) => {
  const z = math.bignumber(v) as math.BigNumber;
  (z as any).toJSON = () => {
    return Number(math.larger(100, z) ? math.round(z,2) : math.round(z,4)).toFixed(4);
  }
  return z;
}
```

but for some reason, it's still stringifying it to:

```js
{"mathjs":"BigNumber","value":"42500"}
```

my goal is to stringify it to a number:

```js
42500
```

**Comment by josdejong** [comment]
> but for some reason, it's still stringifying it to:
> 
> ```
> {"mathjs":"BigNumber","value":"42500"}
> ```

The reason for that is that mathjs serializes all data types into this format. This way, it can correctly deserialize the value into a BigNumber again. If you serialize it to `42500`, you can't know if it originally was a `number` or a `BigNumber` for example. See docs: https://mathjs.org/docs/core/serialization.html

You can override the `toJSON` as follows: 

```js
math.BigNumber.prototype.toJSON = function () {
  return this.toNumber()
}
```
Be careful though, BigNumbers that cannot be represented as a regular `number` will lose numeric information:

```js
math.bignumber('12300000000000000077').toNumber()
// 12300000000000000000
```
